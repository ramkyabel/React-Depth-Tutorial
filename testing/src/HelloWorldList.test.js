import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import HelloWorldList from './HelloWorldList';
import AddGreeter from './AddGreeter';
import HelloWorld from './HelloWorld';

describe (HelloWorldList, () => {
	const component = shallow (
		<HelloWorldList />
	);

	it ('renders and matches our snapshot', () => {
		const component = renderer.create (
			<HelloWorldList />
		);
		const tree = component.toJSON ();
		expect (tree).toMatchSnapshot ();
	});

	it ('contains an AddGreeter subcomponent', () => {
		expect (component.find (AddGreeter)).toHaveLength (1);
	});

	it ('contains the same number of HelloWorld components as greetings', () => {
		const helloWorlds = component.find (HelloWorld).length;
		const greetings   = component.state('greetings').length;
		// console.log(component);
		expect (helloWorlds).toEqual (greetings);
	});

	it ('adds another greeting when the add greeting function is called', () => {
		const greetingsBefore = component.state ('greetings').length;
		const componentsBefore = component.find (HelloWorld).length;

		component.instance().addGreeting	('Added');

		const greetingsAfter = component.state ('greetings').length;
		const componentsAfter = component.find (HelloWorld).length;

		expect(greetingsAfter).toBeGreaterThan(greetingsBefore);
		expect(componentsAfter).toBeGreaterThan(componentsBefore);

	});	

	it ('removes a greeting from the list when the remove greeting functions' +
		'is being called.', () => {
			const before = component.find (HelloWorld).length;
			const greetRemove = 
				component.state('greetings')[component.state('greetings').length - 1];
			component.instance ().removeGreeting(greetRemove);
			const after = component.find (HelloWorld).length;
			expect (after).toBeLessThan (before);
	});
});