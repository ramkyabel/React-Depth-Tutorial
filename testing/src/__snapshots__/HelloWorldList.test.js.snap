// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class HelloWorldList extends _react.Component {
	constructor(props) {
		super(props);
		this.state = { greetings: ['Jim', 'Sally', 'Bender'] };
		this.addGreeting = this.addGreeting.bind(this);
		this.removeGreeting = this.removeGreeting.bind(this);
	}
	addGreeting(newName) {
		this.setState({ greetings: [...this.state.greetings, newName] });
	}
	removeGreeting(removeName) {
		const filteredGreetings = this.state.greetings.filter(name =>
		removeName !== name);
		this.setState({ greetings: filteredGreetings });
	}
	renderGreetings() {
		return this.state.greetings.map(name =>
		_react2.default.createElement(_HelloWorld2.default, {
			key: name,
			name: name,
			removeGreeting: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 23 }, __self: this }));

	}
	render() {
		return (
			_react2.default.createElement('div', { className: 'HelloWorldList', __source: { fileName: _jsxFileName, lineNumber: 31 }, __self: this },
				_react2.default.createElement(_AddGreeter2.default, { addGreeting: this.addGreeting,
					removeGreeting: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 32 }, __self: this }),

				this.renderGreetings()));


	}} renders and matches our snapshot 1`] = `
<div
  className="HelloWorldList"
>
  <div
    className="AddGreeter"
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
      
    <button
      className="add"
      onClick={[Function]}
    >
       Add 
    </button>
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    Hello
    , 
    Jim
    <br />
    <button
      className="frenchify"
      onClick={[Function]}
    >
       Frenchify 
    </button>
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Me! 
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    Hello
    , 
    Sally
    <br />
    <button
      className="frenchify"
      onClick={[Function]}
    >
       Frenchify 
    </button>
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Me! 
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    Hello
    , 
    Bender
    <br />
    <button
      className="frenchify"
      onClick={[Function]}
    >
       Frenchify 
    </button>
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Me! 
    </button>
  </div>
</div>
`;
